views/js/qtiCreator/widgets/interactions/hottextInteraction/states/Question.js

```diff
@@ -104,22 +104,31 @@ define([
 
     HottextInteractionStateQuestion.prototype.initForm = function initForm(){
 
-        var _widget = this.widget,
-            $form = _widget.$form,
-            interaction = _widget.element,
-            callbacks;
-
-        $form.html(formTpl({
-            maxChoices : interaction.attr('maxChoices'),
-            minChoices : interaction.attr('minChoices'),
-            choicesCount : _.size(interaction.getChoices())
-        }));
+        var widget      = this.widget;
+        var $form       = widget.$form;
+        var interaction = widget.element;
+        var callbacks;
+
+        $form.html(formTpl());
+
+        //controls min and max choices
+        minMaxComponentFactory($form.find('.min-max-panel'), {
+            min : { value : _.parseInt(interaction.attr('minChoices')) || 0 },
+            max : { value : _.parseInt(interaction.attr('maxChoices')) || 0 },
+            upperThreshold : _.size(interaction.getChoices())
+        }).on('render', function(){
+            var self = this;
+            widget.on('choiceCreated choiceDeleted', function(data){
+                if(data.interaction.serial === interaction.serial){
```

Comment:
It is an existing bug: transform 2 pieces of text into a hottext, set max to 2, remove 1 hottext, max is still 2, add 3 hottext, the max cannot go beyond 2. 
And there are some error in the console.